cmd/operator/main.go

```diff
@@ -80,7 +82,25 @@ func main() {
 
 	ns, found := os.LookupEnv(NamespaceEnvVar)
 	if !found {
-		setupLog.Error(fmt.Errorf("%s must be set", NamespaceEnvVar), "unable to get namespace")
+		setupLog.Error(fmt.Errorf("%s environment variable must be set (commonly set automatically in every Pod)", NamespaceEnvVar), "unable to get namespace")
+		os.Exit(1)
+	}
+
+	// must have openshift versions specified on the operator
+	// this prevents us from having to include the full json in source
+	// ie. this should ALWAYS be specified on the CSV, until a proper
+	// API is provided for it
+	// I think it's reasonable to check that the OPENSHIFT_VERSIONS is
+	// legit before we go passing it down to the assisted-service deployment
+	// and letting it fail there.
+	var openshiftVersionsMap models.OpenshiftVersions
+	openshiftVersions, found := os.LookupEnv(controllers.OpenshiftVersionsEnvVar)
+	if !found || openshiftVersions == "" {
+		setupLog.Error(fmt.Errorf("%s environment variable must be set (commonly set automatically in every Pod) to a non-empty value.", controllers.OpenshiftVersionsEnvVar), "unable to get OpenShift Versions")
```

Comment:
NIIT the error message seems not related "commonly set automatically in every Pod"

Reply:
That's specifically referring to what we've done with the manager.yaml so that it's automatically included in the ClusterServiceVersion. I'll spend some time thinking about if there are ways to make that more clear.